src/server/plugins/initialize.js

```diff
@@ -40,6 +40,10 @@ module.exports = async function (kbnServer, server, config) {
     path.pop();
   };
 
+  for (let plugin of plugins) {
+    await plugin.preInit();
```

Comment:
For the current use case, it doesn't, but I could see why that might be necessary. What are your thoughts?